src/notification/NotificationController.ts

```diff
@@ -0,0 +1,266 @@
+import type { Patch } from 'immer';
+import { nanoid } from 'nanoid';
+
+import { hasProperty } from '../util';
+import { BaseController } from '../BaseControllerV2';
+
+import type { RestrictedControllerMessenger } from '../ControllerMessenger';
+
+/**
+ * @type NotificationState
+ * @property notifications - Stores existing notifications to be shown in the UI
+ */
+export type NotificationState = {
+  notifications: Record<string, Notification>;
+};
+
+export type Notification = {
+  id: string;
+  type: NotificationType;
+  origin: string;
+  date: number;
+  read: boolean;
+  message: string;
+};
+
+export enum NotificationType {
+  Native = 'native',
```

Comment:
I was going to re-state here the same opinion I shared in Slack, which is that native notifications aren't similar enough to in-app notifications to warrant putting them in the same controller. But I've looked into it and I have changed my mind.

I thought that there was no way or reason to track whether a native notification has been dismissed or read, but I was wrong on both counts. There is an event for both cases, and there is a reason to track it as well (to inform snaps what the current status of the notification is).

Reply:
Yeah, at this point though we don't track any of that, so we'll need to expand upon or native notification implementation for that. Out of scope for this PR imo 😄

Reply:
Removed